staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/validating/dispatcher.go

```diff
@@ -100,11 +100,25 @@ func (d *validatingDispatcher) Dispatch(ctx context.Context, attr admission.Attr
 	}
 
 	wg := sync.WaitGroup{}
-	errCh := make(chan error, len(relevantHooks))
+	errCh := make(chan error, 2*len(relevantHooks)) // double the length to handle extra errors for panics in the gofunc
 	wg.Add(len(relevantHooks))
 	for i := range relevantHooks {
 		go func(invocation *generic.WebhookInvocation, idx int) {
-			defer wg.Done()
+			ignoreClientCallFailures := false
+			defer func() {
+				defer wg.Done()
```

Comment:
This was neat.  As a different defer, this produced a datarace with the channel closure down below.  A good example of the utility of unit test coverage that I wrote in https://github.com/kubernetes/kubernetes/pull/108746, but I don't know how to set `utilruntime.ReallyCrash=false` without having process-wide impact.

Reply:
really? this didn't execute panic recovery before waitgroup closure?

```
defer wg.Done()
defer func(){ ... panic recovery handling ... }()
```

Reply:
> As a different defer, this produced a datarace with the channel closure down below

that is ... pretty surprising to me

Reply:
When you get it wrong go test k8s.io/apiserver/pkg/admission/plugin/webhook/validating -run "TestValidatePanicHandling" --race produces

--- FAIL: TestValidatePanicHandling (0.32s)
testing.go:1152: race detected during execution of test

But I haven't figured out how to merge the tests and depending on, "be careful to add these defers in the right order" wasn't clearly better to me than "defer inside of defer ensures correct ordering". I'm flexible about factorization if your preference is to rely on the order in which defers are added, which is apparently called in reverse order (not the forward order that I would naively expect).

from the proof PR: https://github.com/kubernetes/kubernetes/pull/108746